File: calculadora_switch.c
#include <stdio.h>
float divi, multi, suma, resta, num1, num2;
int opcion;
int main()
{
    printf("Ingresa el primer numero\n");
    scanf("%f", &num1);
    printf("Ingrese el segundo numero\n");
    scanf("%f", &num2);

    printf("Ingrese la operacion que quiera realizar\n");
    printf("1. +\n");
    printf("2. -\n");
    printf("3. *\n");
    printf("4. /\n");
    scanf("%d", &opcion);

    switch (opcion)
    {
    case 1:
        suma = num1 + num2;
        printf("La suma de su numero es %.2f", suma);
        break;
    case 2:
        resta = num1 - num2;
        printf("La resta de su numero es %.2f", resta);
        break;
    case 3:
        multi = num1 * num2;
        printf("La multiplicacion de su numero es %.2f", multi);
        break;
    case 4:
    if ("num2 == 0"){
        printf("Error. No se puede dividir entre 0\n");
    }
        divi = num1 / num2;
        printf("La division de su numero es %.2f", divi);
        break;
    default:
            printf("Opcion invalida");
        break;
    }
    return 0;
}

File: calculadora_switch.py
print("-------------------\n")
print("CALCULADORA\n")
print("-------------------\n")
num1 = float(input("Número 1: "))
num2 = float(input("Número 2: "))

print("Seleccione la operación a realizar:")
print("1. Suma")
print("2. Resta")
print("3. Multiplicación")
print("4. División")
operacion = int(input("Ingrese la operacion que quiera realizar: "))

match operacion:
    case 1:
        resultado = num1 + num2
        print(f"La suma de su número es: {resultado}")
    case 2:
        resultado = num1 - num2
        print(f"La resta de su número es: {resultado}")
    case 3:
        resultado = num1 * num2
        print(f"La multiplicación de su número es: {resultado}")
    case 4:
        if num2 != 0:
            resultado = num1 / num2
            print(f"La división de su número es: {resultado}")
        else:
            print("Error: No se puede dividir entre cero.")
    case _:
        print("Operación no válida. Por favor, seleccione una opción entre 1 y 4.")